org.eclipse.ui.console/src/org/eclipse/ui/internal/console/IOConsolePartitioner.java

```diff
@@ -1465,4 +1478,65 @@ private void checkPartitions() {
 			Assert.isTrue(knownInputPartitions.isEmpty());
 		}
 	}
+
+	public void showPreviousCommand() throws ExecutionException {
+		String command = _commandHistory.prevCommand();
+		replaceIncompleteUserInputWith(command);
+	}
+
+	public void showNextCommand() throws ExecutionException {
+		String command = _commandHistory.nextCommand();
+		replaceIncompleteUserInputWith(command);
+	}
+
+	/**
+	 * Replace the last input the user entered with the given command. If the last
+	 * user input is not the last partition, we append an input partition.
+	 *
+	 * @param command Replaces the last user input.
+	 * @throws ExecutionException An error occurred attempting to insert history.
+	 */
+	@SuppressWarnings("resource")
+	public void replaceIncompleteUserInputWith(String command) throws ExecutionException {
+		if (!enableCommandHistory) {
+			return;
+		}
+
+		int offset;
```

Comment:
i'd explicitely initialize this to 0 to convey te information that 0 is a legit value when this variable is used in line 1524